MainForm.cs

```diff
@@ -0,0 +1,249 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Windows.Forms;
+using System.Reflection;
+using System.ComponentModel.DataAnnotations;
+
+namespace Task1_GUI
+{
+    public enum Colors
+    {
+        [Display(Name = "#FF5B58")]
+        R = 0,
+
+        [Display(Name = "#FFAC40")]
+        O = 1,
+
+        [Display(Name = "#FFF650")]
+        Y = 2,
+
+        [Display(Name = "#8DFF70")]
+        G = 3,
+
+        [Display(Name = "#97FFF3")]
+        C = 4,
+
+        [Display(Name = "#7F8CFF")]
+        B = 5,
+
+        [Display(Name = "#A978FF")]
+        P = 6
+    }
+    public partial class MainForm : Form
+    {
+        public MainForm()
+        {
+            InitializeComponent();
+        }
+        public void readData(ref bool isDataCorrect, int combLen, string[] comb, List<int> nomOfCards, TextBox textBox, List<string> colors, List<int> priotityOfColors)
+        {
+            const int nomPosition = 0, colorPosition = 2;
+            for (int i = 0; i < combLen; i++)
+            {
+                int nom;
+                bool result1 = int.TryParse(Convert.ToString(comb[i][nomPosition]), out nom);
```

Comment:
Let `result1` will have the name `isCardNominalCorrect`. All bool variables should have meaningful names. Like isDataValid, or needToSaveCollection.